test/jit/test_pdt.py

```diff
@@ -405,3 +407,32 @@ def test_model_with_args(a, m):
         user_class = ClassWithArgs(False)
         scripted_fn = torch.jit._script_pdt(test_model_with_args, example_inputs=[(10, user_class, ), (10.9, user_class, ), ])
         self.assertEqual(scripted_fn(100, ClassWithArgs(True), ), test_model_with_args(100, ClassWithArgs(True)))
+
+
+    def test_function_scripted_many_times(self):
+        def test_script_add_of_two_var(x, y):
+            return x + y
+
+        make_global(test_script_add_of_two_var)
+
+        scripted_fn = torch.jit._script_pdt(test_script_add_of_two_var, example_inputs=[(1, 2), ])
+        self.assertEqual(scripted_fn(10, 20), test_script_add_of_two_var(10, 20))
+        scripted_fn = torch.jit._script_pdt(test_script_add_of_two_var, example_inputs=[(1.6, 2.9), ])
+        self.assertEqual(scripted_fn(-1.8, 20.14), test_script_add_of_two_var(-1.8, 20.14))
+        scripted_fn = torch.jit._script_pdt(test_script_add_of_two_var, example_inputs=[(89, 27), ])
```

Comment:
Does recompilation make previously compiled function (with different signature) unavailable? If so, that can be a problem because there may be another live module that references previously compiled function.

Reply:
For example:
```
import torch
from typing import Any, List, Tuple

def fn(x, y):
    return x + y

# Float inputs
s1 = torch.jit._script_pdt(fn, example_inputs=[(1.2, 3.4), ])
print("Float input : ", s1(4.5, 7.1))

# Int Inputs
s2 = torch.jit._script_pdt(fn, example_inputs=[(1, 2), ])
print("Int input : ", s2(2, 3))

print("Float input again : ", s1(4.9, 8.1))
```
Output
```
Float input :  11.6
Int input :  5
Float input again :  13.0
```
So here, both scripted functions `s1` and `s2` are available and return the correct output. Is this what you mean ?

Reply:
Discussed offline:

Maybe try creating two modules that calls `fn` in `forward`s. But they use different types for `fn`'s input.

Reply:
Tried this and I did not see any issue. I have added a new test case `test_two_nn_modules_scripting_same_function` and it works fine.

Reply:
I pulled this PR to my local repo, merged the API on top of it, but the issue persists. Looked into it a little bit, when compiling second module, the only function available for schema matching is int-version of `test_add_fn`.

See my local change here:
https://github.com/pytorch/pytorch/pull/59518